src/oumi/quantize/awq_quantizer.py

```
@@ -0,0 +1,189 @@
+from oumi.quantize.utils import format_size, get_directory_size
+from oumi.utils.logging import logger
+
+# AWQ configuration defaults
+AWQ_DEFAULTS = {
+    "calibration_dataset": "pileval",
+    "calibration_split": "train",
+    "calibration_text_column": "text",
+    "max_calibration_seq_len": 512,
+    "duo_scaling": True,
+    "apply_clip": True,
+    "n_parallel_calib_samples": None,
+}
+
+
+class AwqQuantization(BaseQuantization):
+    """AWQ (Activation-aware Weight Quantization) implementation.
+
+    This class handles AWQ quantization with support for simulation mode
+    when AWQ libraries are not available.
+    """
+
+    supported_methods = ["awq_q4_0", "awq_q4_1", "awq_q8_0", "awq_f16"]
+    supported_formats = ["pytorch"]
+
+    def __init__(self):
+        """Initialize AWQ quantizer."""
+        self._awq = importlib.util.find_spec("autoawq")
+
+    @override
+    def raise_if_requirements_not_met(self):
+        """Check if AWQ dependencies are available."""
+        if self._awq is None:
+            raise RuntimeError(
+                "AWQ quantization requires autoawq library.\n"
+                "Install with: `pip install autoawq`\n"
+            )
+
+        if not torch.cuda.is_available():
+            raise RuntimeError(
+                "AWQ quantization requires a GPU. "
+                "Please use a machine with at least 1 GPU."
+            )
+
+    @override
+    def quantize(self, config: QuantizationConfig) -> dict[str, Any]:
+        """Main quantization method for AWQ.
+
+        Args:
+            config: Quantization configuration
+
+        Returns:
+            Dictionary containing quantization results
+        """
+        self.validate_config(config)
+
+        logger.info("Starting AWQ quantization pipeline...")
+
+        # Step 1: AWQ quantization
+        awq_result = self._quantize_model_with_awq(config)
+        awq_model_path = awq_result["awq_model_path"]
+
+        # Step 2: Save as PyTorch format
```

Let's throw an exception (NotImplementedError) if config.format_type != pytorch since it's the only one we support